network_model/network modeling/network_analysis.py

```diff
@@ -0,0 +1,270 @@
+
+
+def tower_distances_from_gps_data(gps_data, cell_coords, city_distances, city_data):
+    ##get the distances of all towers from gps_data and return a dictionary where an index of the person is the key of the distances
+    ##dictionary and the value is the ten closest towers to them (the number of towers can be changed by adding more values to closest_towers)
+    distances = dict()
+    index = 0
+    closest_cities_dict = dict()
+    for person in gps_data:
+        if person[5] not in closest_cities_dict.keys():
+            closest_cities = [[" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")]]
+            for city in city_distances.keys():
+                if city.lower() == person[5].lower():
+                    cities = city_distances[city]
+                    for x in range(0, len(cities) - 1):
+                        dist = cities[x + 1]
+                        for y in range(0, 5):
+                            if dist < closest_cities[y][1]:
+                                closest_cities[y] = [cities[x], cities[x + 1]]
+                                break
+                        x += 1
+            closest_cities_dict[person[5]] = closest_cities
+        closest_cities = closest_cities_dict[person[5]]
+        closest_towers = [[" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")], [" ", float("inf")]]
+        try:
+            all_available_towers = city_data[person[5].lower()][3]
+        except:
+            continue
+        for x in range(0, len(closest_cities)):
+            try:
+                all_available_towers.append(city_data[closest_cities[x][0]][3])
+            except:
+                continue
+        for cell in cell_coords.keys():
+            if cell in all_available_towers:
+                for x in range(0, 10):
+                    dist = coords_to_metric(float(person[0]), float(person[1]), float(cell_coords[cell][0]), float(cell_coords[cell][1]))
+                    if dist < closest_towers[x][1]:
+                        closest_towers[x] = [cell, dist]
+                        break
+        distances[index] = closest_towers
+        index += 1
+    return distances
+
+
+def total_towers_within_range(gps_data, cell_coords):
+    ##return a dictionary of all towers within a certain range to each person where the index of the person is the key and the value is the list of towers
+    towers = dict()
+    index = 0
+    for person in gps_data:
+        count = 0
+        for cell in cell_coords:
+            dist = coords_to_metric(float(person[0]), float(person[1]), float(cell_coords[cell][0]), float(cell_coords[cell][1]))
+            if dist <= 10000:
+                count += 1
+        towers[index] = count
+        index += 1
+    print(towers)
+    print(len(cell_coords))
+    print(len(towers))
```

Comment:
Looks like this should be returning stuff [CHANGE REQUESTED]

Reply:
Done.